resources/assets/js/src/components/chart/Chart.tsx

```diff
@@ -209,62 +209,49 @@ export default class Chart extends React.Component<IProps, IState> {
     }
 
     private loadData(dateFrom: string, dateTo: string, direction: string = null): void {
-        let url = this.props.url;
+        const url = this.createRequestURL(dateFrom, dateTo);
         let dataMeta;
 
-        if (dateFrom !== null && dateTo !== null) {
-            url = `${url}/?start_date=${dateFrom}&end_date=${dateTo}`;
-        }
-
         axios.get(url).then((response: any) => {
-            const newData = [];
-            const length = this.state.data.length > 0 ? response.data.length - 1 : response.data.length;
-
-            for (let i = 0; i < length; i++) {
-                newData.push({
-                    date: response.data[i].date,
-                    value: response.data[i][this.props.columnName],
-                });
-            }
-
-            if (newData[0].date === this.state.initialDate) {
+            const newData = this.processResponse(response);
+            if (!newData.length || newData[0].date === this.state.initialDate) {
                 return;
             }
 
             if (direction) {
-                const data = [...this.state.data];
-
-                switch (direction) {
-                    case "plus":
-                        data.push(...newData);
-                        break;
-                    case "minus":
-                        data.unshift(...newData);
-                        break;
-                    default:
-                        return null;
-                }
-
+                let data = (direction === "plus") ? [...this.state.data, ...newData] : [...newData, ...this.state.data];
+                data = ArrayUtil.removeDuplicities(data);
```

Comment:
why there are duplicities?

Reply:
This might be also fixed on backend side

Reply:
because API returns duplicite data - if i load data for 12:30 - 13:30 and then 13:30 - 14:30, there will be two objects for time 13:30... this is the most effective way to handle this problem, because i don't know time between records (so i can't remove this time from interval) and map + removeDuplicities is more effective then handling exceptions for different cases (tested)